### PR TITLE
ROCANA-8849: Update posting format fork to remove checksum

### PR DESCRIPTION
**Pull request by bradcupit**

### CODE REVIEW

File: src/main/java/com/rocana/lucene/codec/v1/RocanaSearchCodecV1.java [hunk 1]
```diff
@@ -54,4 +92,23 @@ Codec getWrappedCodec() {
     return delegate;
   }
 
+  /**
+   * This codec wraps the {@link Lucene54Codec}, which itself
+   * returns a {@link PerFieldPostingsFormat} that wraps
+   * {@link RocanaLucene50PostingsFormat}.
```
Comment (mpeterson0x0) [inline src/main/java/com/rocana/lucene/codec/v1/RocanaSearchCodecV1.java]: this reference is wrong - should point to the Lucene54Codec?  Maybe not, the comment is actually a little hard to follow. Perhaps rephrase?

File: src/main/java/com/rocana/lucene/codec/v1/RocanaSearchCodecV1.java [hunk 2]
```diff
@@ -54,4 +92,23 @@ Codec getWrappedCodec() {
...
+  @Override
+  public PostingsFormat postingsFormat() {
+    return new PerFieldPostingsFormat() {
+
+      @Override
+      public PostingsFormat getPostingsFormatForField(String field) {
+        return forkedPostingsFormat;
+      }
+    };
+  }
```
Comment (mpeterson0x0) [inline src/main/java/com/rocana/lucene/codec/v1/RocanaSearchCodecV1.java:130]: This implementation creates and returns a new `PerFieldPostingsFormat` for every invocation, whereas the Lucene code constructs a single object and always returns that.

This is the code in `Lucene54Codec`:
```
  @Override
  public final PostingsFormat postingsFormat() {
    return postingsFormat;
  }
```

If there is a DGR for doing this, that should be documented. Otherwise, let's follow the Lucene model and create fewer objects.

**Comment by mpeterson0x0** [comment]
In which ticket are you going to remove the call to `CodecUtil.checksumEntireFile`? As far as I can tell, after these three tickets it is still there in the `RocanaBlockTreeTermsReader`.

**Comment by bradcupit** [comment]
Wow that's embarrassing. I'd made the change on some older versions of the files but didn't move it here, to the final version. EXCELLENT catch!

This PR is the most appropriate one to put it in.

File: src/main/java/com/rocana/lucene/codec/v1/RocanaSearchCodecV1.java (hunk 2 above)
Comment (bradcupit) [reply src/main/java/com/rocana/lucene/codec/v1/RocanaSearchCodecV1.java:130]: I totally agree. Excellent catch!

File: src/main/java/com/rocana/lucene/codec/v1/RocanaSearchCodecV1.java (hunk 1 above)
Comment (bradcupit) [reply src/main/java/com/rocana/lucene/codec/v1/RocanaSearchCodecV1.java]: That comment is definitely hard to understand. Improvement to follow.

File: src/main/java/com/rocana/lucene/codec/v1/RocanaSearchCodecV1.java [hunk 3]
```diff
@@ -32,15 +36,68 @@
...
+ * We don't want to customize Lucene's on-disk format. We just want
+ * to stop Lucene from checksumming an entire file when there's no
+ * benefit. The Lucene checksum is redundant since we store indexes
+ * on HDFS, which also checksums. The Lucene checksum happens at an
+ * inopportune time as we're trying to open a Rocana Search Searcher,
+ * which may happen during a rebalance, and we'd rather eliminate
+ * that penalty altogether or at least delay it until the rebalance
+ * finishes.
+ * 
+ * To accomplish that we register this class as the default Lucene
```
Comment (mpeterson0x0) [inline src/main/java/com/rocana/lucene/codec/v1/RocanaSearchCodecV1.java]: I don't think we "register as the default". If we did the classpath overriding then we would. Lucene has a method `Codec.getDefault()` and that will still return Lucene54Codec (I think - double check me on that). Instead, we "register an alternative Codec through the Lucene Codec SPI system" is probably how I would phrase it.

Comment (mpeterson0x0) [reply src/main/java/com/rocana/lucene/codec/v1/RocanaSearchCodecV1.java]: We should probably also mention that this codec has its own unique "name" that will be written to the segment info files, rather than the Lucene54 name, even though we still use the Lucene54Codec under the hood for most operations.

**Review by mpeterson0x0 — APPROVED** [review]
+1

File: src/main/java/com/rocana/lucene/codec/v1/RocanaSearchCodecV1.java (hunk 3 above)
Comment (bradcupit) [reply src/main/java/com/rocana/lucene/codec/v1/RocanaSearchCodecV1.java]: Good suggestions, I'll write it up!

Comment (bradcupit) [reply src/main/java/com/rocana/lucene/codec/v1/RocanaSearchCodecV1.java]: done and pushed
